src/proteus/dswx_hls.py

```diff
@@ -1038,15 +1041,38 @@ def create_landcover_mask(copernicus_landcover_file,
     worldcover_gdal_ds = gdal.Open(worldcover_file, gdal.GA_ReadOnly)
     worldcover_metadata = worldcover_gdal_ds.GetMetadata()
     del worldcover_gdal_ds
-    worldcover_datetime_format = '%Y-%m-%dT%H:%M:%SZ'
-    worldcover_time_start = datetime.strptime(worldcover_metadata['time_start'],
-                                              worldcover_datetime_format)
-    worldcover_time_end = datetime.strptime(worldcover_metadata['time_end'],
-                                            worldcover_datetime_format)
 
-    # the Worldcover dataset year is extracted from the average date times
-    worldcover_time_range = (worldcover_time_end - worldcover_time_start)
-    year = (worldcover_time_start + worldcover_time_range / 2.0).year - 2000
+    if ('time_start' in worldcover_metadata.keys() and
+            'time_end' in worldcover_metadata.keys()):
+        worldcover_datetime_format = '%Y-%m-%dT%H:%M:%SZ'
+        worldcover_time_start = datetime.strptime(worldcover_metadata['time_start'],
+                                                  worldcover_datetime_format)
+        worldcover_time_end = datetime.strptime(worldcover_metadata['time_end'],
+                                                worldcover_datetime_format)
+
+        # the Worldcover dataset year is extracted from the average date times
+        worldcover_time_range = (worldcover_time_end - worldcover_time_start)
+        year = (worldcover_time_start + worldcover_time_range / 2.0).year
+        logger.info('ESA WorldCover map year: {year}'
+                    '(source: WorldCover file metadata)')
+
+    else:
+        logger.warning('WARNING Could not read the ESA WorldCover 10m metadata'
+                       ' fields `time_start` and/or `time_end`')
+        year_start = 2000
+        year_end_next = 2100
+        for year in range(year_start, year_end_next)
+            if str(year) in worldcover_file_description:
+                logger.info('ESA WorldCover map year: {year}'
+                            '(source: WorldCover file description)')
+                break
+        else:
+            year = 2000
+            logger.warning('WARNING Could not infer the ESA WorldCover 10m'
+                           ' data year from the WorldCover file description.'
+                           ' Considering it as 0 (2000)')
+
+    year -= 2000
```

Comment:
Re: year -- Documenting here a summary our offline discussion from yesterday:

After this release, it would be best to remove all references to the year in the LAND data values themselves, and instead simply use the metadata field `WORLDCOVER_SOURCE` to document the year of the WorldCover input source for the users.

Based on that change, it would be really nice to update the data value classifications to something simpler, such as:

```
0: [Everything else]
1: Water, Wetland, Mangrove Forest
2: Low Intensity Developed
3: High Intensity Developed
4: A variety of  Forest Classes
255: Fill value (no data)
```